docs/guide/module-configuration-patterns.md

```diff
@@ -0,0 +1,332 @@
+# Module Extension and Improvement
+
+One core feature of ApostropheCMS is its module system that allows you to customize and enhance functionality without modifying core code. This approach keeps your project maintainable and upgrade-friendly while giving you the flexibility to tailor the CMS to your needs. You will see this information partially repeated in several areas of our documentation including in the [reference section](/reference/module-api/module-overview.html#extend) because we feel it is so important for developers to understand.
+
+## Understanding Module Customization in ApostropheCMS
+
+When working with ApostropheCMS, it's important to understand that customizations should be made through files in your project's `/modules` folder rather than by directly modifying files in the `node_modules` directory.
+
+Modifying files in `node_modules` can create significant problems:
+- Changes are lost when you run `npm install` or `npm update`
+- Customizations won't carry over when deploying to different environments
+- Your changes won't be tracked in version control
+
+Instead, ApostropheCMS is designed to let you customize modules by creating corresponding files in your project's `/modules` folder. This way, your changes are:
+- Preserved during updates
+- Properly tracked in version control
+- Correctly deployed across environments
+- Applied consistently throughout your application
+
+## The Three Types of Module Customization
+
+ApostropheCMS provides three distinct ways to customize module behavior, each with its own purpose:
+
+### 1. Module Configuration
+
+**Use for:** Adjusting existing behavior using options that have been specifically built into modules.
+
+Configuration provides a way to set values for options that are already designed into the modules. You can only configure options that the module explicitly supports in this way, not create new ones. You can apply configuration in these places:
+
+1. In your `app.js` file when registering modules:
+
+
+<AposCodeBlock>
+
+```javascript
+export default {
+  modules: {
+    '@apostrophecms/image': {
+      options: {
+        // Simple configuration settings
+        sizes: '(min-width: 600px) 45vw, (min-width: 1140px) 530px'
+      }
+    }
+  }
+}
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+</AposCodeBlock>
+
+2. In the `options` object when creating or customizing a module at the project level:
```

Comment:
I think it might be better to just say you can always set options whether you're creating a new module with `extend` or just configuring an included one like `@apostrophecms/page`.

Reply:
Changed.